src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -310,6 +403,41 @@ func (h *PromWriteHandler) write(
 	return h.downsamplerAndWriter.WriteBatch(ctx, iter, opts)
 }
 
+func (h *PromWriteHandler) forward(
+	ctx context.Context,
+	request prometheus.ParsePromCompressedRequestResult,
+	target PromWriteHandlerForwardTargetOptions,
+) error {
+	method := target.Method
+	if method == "" {
+		method = http.MethodPost
+	}
+
+	req, err := http.NewRequest(target.Method, target.URL,
+		bytes.NewReader(request.CompressedBody))
+	if err != nil {
+		return err
+	}
+
+	timeout := defaultForwardingTimeout
+	if h.forwarding.Timeout > 0 {
+		timeout = h.forwarding.Timeout
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	resp, err := http.DefaultClient.Do(req.WithContext(ctx))
```

Comment:
I'd be careful with this, you have to do some configuring to Go's http client to get "sensible" behavior. For example I think it will limit you to to connections per host so if you're sending a lot of traffic to a load balancer you can end up getting bottlenecked by constantly creating new connections pretty quickly:

http://tleyden.github.io/blog/2016/11/21/tuning-the-go-http-client-library-for-load-testing/

Reply:
Yeah I'm going to pull in our tuned HTTP client in a followup change, I need to move it from some benchmarking code:
https://github.com/m3db/m3/blob/master/src/query/benchmark/write/main/benchmarker.go#L62-L97